File: backend/src/main/java/com/distributed/entity/Node.java
package com.distributed.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Entity
@Table(name = "nodes")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Node {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, length = 255)
    private String host;

    @Column(nullable = false)
    private Integer port;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private NodeStatus status = NodeStatus.ACTIVE;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    protected void onCreate() {
        Instant now = Instant.now();
        this.createdAt = now;
        this.updatedAt = now;
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = Instant.now();
    }

    public enum NodeStatus {
        ACTIVE, INACTIVE, MAINTENANCE
    }
}

File: frontend/src/types/node.ts
export type NodeStatus = 'ACTIVE' | 'INACTIVE' | 'MAINTENANCE'

export interface Node {
  id: number
  name: string
  host: string
  port: number
  status: NodeStatus
  createdAt: string
  updatedAt: string
}

export interface NodeCreateInput {
  name: string
  host: string
  port: number
  status?: NodeStatus
}

export interface NodeUpdateInput {
  name?: string
  host?: string
  port?: number
  status?: NodeStatus
}
